packages/block-editor/src/components/inserter/index.js

```diff
@@ -105,10 +119,67 @@ class Inserter extends Component {
 
 export default compose( [
 	withSelect( ( select, { rootClientId } ) => {
-		const { hasInserterItems } = select( 'core/block-editor' );
-
+		const {
+			hasInserterItems,
+			__experimentalHasOnlyOneAllowedBlockType,
+			__experimentalGetTheOnlyAllowedBlockType,
```

Comment:
I was wondering why two selectors are needed? Could an `__experimentalGetAllowedBlocks` selector serve both purposes?

Reply:
yes, I thought it was better to have a list getter and a boolean selector but I also tried your idea and it is even better :)